src/parser.ts

```diff
@@ -4178,8 +4169,24 @@ export class Parser extends DiagnosticEmitter {
         case Token.BAR:
         case Token.CARET:
         case Token.AMPERSAND_AMPERSAND:
-        case Token.BAR_BAR: {
-          let next = this.parseExpression(tn, nextPrecedence + 1);
+        case Token.BAR_BAR:
+          ++nextPrecedence;
```

Comment:
Most code places single statements inline and otherwise uses braces, with no hard requirement to not use braces. Two consistent options:

```suggestion
        case Token.BAR_BAR: ++nextPrecedence;
```
or
```suggestion
        case Token.BAR_BAR: {
          ++nextPrecedence;
        }
```